### ISSUE TITLE
Error with Dorado Resume

### ISSUE THREAD

**Issue by lucky5sugar** [nanoporetech/dorado#454]
To the development team,

Thank you for your continued support. 
When passing a large interrupted BAM file (approximately 90 GB) as "incomplete.bam" using `dorado basecaller --resume-from incomplete.bam > calls.bam`, I encountered an error where a large number of dots were printed on the screen, and the analysis did not progress. 
When I tried reducing the size of the BAM file, with around 100 reads, I did not encounter the error, but when the file contained over 1000 reads, the same error occurred.
I have tried both dorado versions 0.3.4 and 0.4.2 (linux-x64), but the same error occurred. 

I would appreciate it if I could receive a response regarding this matter.

Best regard

**Comment by tijyojwad** [comment]
Hi @lucky5sugar - thank you for reporting the issue. I will take a look at get back to you on this. Can you post what you mean by `large number of dots`? Was it just a bunch of periods?

**Comment by tijyojwad** [comment]
Hi @lucky5sugar - The dots you are seeing are part of a progress bar visualization indicating that the incomplete bam is being processed. It'll look something like
```
[···<=>··············] Resuming from file
```
Once the file has been read (and it could take some time if it's a large number of reads), the progress bar will switch to
```
[█████████████████████████▏    ] 83% [00m:00s<00m:00s] 
```
If those bars are not being displayed properly, it could be an issue with the terminal. Are you using any non-standard shell?

**Comment by lucky5sugar** [comment]
Thank you for your response.

You are absolutely right; the periods continue to be output in large quantities. I am running it on a GPU server (Ubuntu 20.04), and I am using a regular bash terminal.

Even if the display is strange, I would be fine with it as long as the analysis progresses normally, but I am worried about the analysis. During the period when the periods are being output, the GPU and memory are hardly utilized (although the CPU appears to be heavily used).

Thank you for your assistance.

**Comment by tijyojwad** [comment]
> During the period when the periods are being output, the GPU and memory are hardly utilized

Yes that's expected, since at that time dorado is iterating through the incomplete input file to find finished records that should be skipped. So it's all CPU based processing. The GPU based basecalling should start right after that completes.

**Comment by tijyojwad** [comment]
Hi @plijnzaad - thanks for reporting the logging issue. When dorado is run in non-verbose mode with `stderr` going to file, I believe we had disabled progress bar update. But seems like there's a regression and we'll need to fix that.

If the `stderr` is not redirected to a file, the `ttys` should look something like this for resume.
![out](https://github.com/nanoporetech/dorado/assets/1127155/a1c4c8a6-e019-4a42-bc59-76d0f330b92f)

Are you able to skip stderr redirection or is that something you need?

**Comment by lucky5sugar** [comment]
In the same situation as @plijnzaad , I have around 150 Gbases of pod5 data (approximately 1TB) that had been running on dorado for about 3 days before the program crashed. It has been about 3 more days since I attempted to resume it, but there has been no change whatsoever. Dots, which should presumably represent the progress bar, continue to be continuously output on the screen.

@tijyojwad,  Is there currently any way to avoid this error?

**Comment by tijyojwad** [comment]
Can you paste what you see in your terminal?

3 days to resume seems waaaay too long. How big is the BAM fie you're resuming from? Is this file on a local disk or on a network drive?

**Comment by lucky5sugar** [comment]
The incomplete bam file is 85 GB on a internal network drive. 
What I see are just bunch of dots.
![dorado-resume-error](https://github.com/nanoporetech/dorado/assets/26211395/dcdbea26-eba8-419a-b4a0-fd8d064f807c)

**Comment by plijnzaad** [comment]
> Hi @plijnzaad - thanks for reporting the logging issue. When dorado is run in non-verbose mode with `stderr` going to file, I believe we had disabled progress bar update. But seems like there's a regression and we'll need to fix that.
> 
> If the `stderr` is not redirected to a file, the `ttys` should look something like this 

But that in itself takes up compute resources, I personally think such progress bars are not useful  and in fact they cost i/o, wasting compute resources

> Are you able to skip stderr redirection or is that something you need?

No, I cannot. And I think in the large majority of cases `dorado basecaller` (etc) are  run non-interactively in the cloud or in an HPC queue. So outputting hundreds of GB worth of logging (`--verbose`) or of dots simply should not be done ...

**Comment by tijyojwad** [comment]
That's clearly a bug. We'll look into this. Can you also share which variant of Linux this is running on?

Ack on the excessive debug spew - we're looking into reducing that already.

**Comment by tijyojwad** [comment]
> The incomplete bam file is 85 GB on a internal network drive.

I suspect the resume feature is limited to some extent by how fast the incomplete BAM can be read. I'll run some benchmarks locally to get a sense of how long it _should_ take and get back to you. As for the dots, we'll get a fix into dorado for this ASAP.

**Comment by plijnzaad** [comment]
> Can you also share which variant of Linux this is running on?

Sorry, missed that, not sure if still useful but it's:

```
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Rocky
Description:	Rocky Linux release 8.9 (Green Obsidian)
Release:	8.9
Codename:	GreenObsidian
```
The machine sports an Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz and a Quadro RTX 6000  GPU

**Comment by tijyojwad** [comment]
Hi @plijnzaad - the [fix](https://github.com/nanoporetech/dorado/commit/0a0711012ad906f94aa6e26c3a6b540e5ccbcc0e) for excessive dots in output log was merged as part of dorado v0.5.0.

I still owe you some numbers of how long to expect for resume. Will have that for you soon

**Comment by tijyojwad** [comment]
@lucky5sugar I tested with a 67GB BAM file. The file was on a local SSD. It took <15 mins to resume from that file, so for your dataset I wouldn't expect it to take much more than that. So I suspect the bottleneck in our case is reading from the network drive.
